### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by mohsentm**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+```
+
+or run in terminal: ```composer require becopay/php-payment-gateway```
+
+
+### Usage
+
+**Create Class**<br>
+First use the library with this namespace `use Becopay\PaymentGateway;`.<br>
+for getting the constructor parameters, you first need to register on becopay website([register now](https://becopay.com/en/io/#api))<br>
```
Comment (jeus) [inline readme.md:23]: @mohsentm this url is false

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    "license": "MIT",
+    "authors": [
+        {
+            "name": "becopay team",
+            "email": "io@becopay.com"
+        }
+    ],
+    "support": {
+      "issues": "https://github.com/becopay/php-payment-gateway/issues"
```
Comment (jeus) [inline composer.json:20]: this url change. @mohsentm

File: readme.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    "description": "test payment",
+    "gatewayUrl": "https://gateway.becopay.com/invoice/POS_29_61a5",
+    "callback": "http://www.your-website.com/invoice?orderid=12324320",
+    "orderId": "12324320"
+}
+```
+
+### License
+
+This package is open-source software licensed under the [MIT license](http://opensource.org/licenses/MIT)
```
Comment (jeus) [inline readme.md:164]: change to Apache 2

File: tests/PaymentGatewayConstructorTest.php [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+class PaymentGatewayConstructorTest extends TestCase
+{
+
+    private $dataSet = array(
+        //test constructor with correct data
+        array(
+            'apiUrl' => 'http://api.com',
+            'gatewayUrl' => 'http://gatwayurl',
+            'apiKey' => 'apikey',
+            'mobile' => '09100000000',
```
Comment (jeus) [inline tests/PaymentGatewayConstructorTest.php:28]: mobile numbers remove

File: tests/PaymentGatewayConstructorTest.php [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+            'apiUrl' => 'https://api.url',
+            'gatewayUrl' => 'https://80.80.80.80', //The parameter is being tested
+            'apiKey' => 'apikey',
+            'mobile' => '09100000000',
+            'isAssertion' => true,
+            'test'=>'Test $gatewayUrl parameter with Ip'
+        ),
+        // Test $gatewayUrl parameter with Ip and port
+        array(
+            'apiUrl' => 'https://api.url',
```
Comment (jeus) [inline tests/PaymentGatewayConstructorTest.php:156]: api url have to convert to variable that load from property file.

File: tests/PaymentGatewayCreateTest.php [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        parent::__construct($name, $data, $dataName);
+        $this->config = new LoadConfig();
+    }
+
+    public function dataSet()
+    {
+        return $dataSet = array(
+            // Test invalid api key
+            array(
+                'apiUrl' => 'http://87.98.188.77:9193',
```
Comment (jeus) [inline tests/PaymentGatewayCreateTest.php:31]: remove this from source code. :|

File: tests/PaymentGatewayCreateTest.php [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+                'mobile' => $this->config->MOBILE,
+                'orderId' => '4988849849',
+                'price' => '54166',
+                'description' => 'test order',
+                'isAssertion' => false,
+                'test' => 'Test invalid api key'
+            ),
+            //Test invalid mobile
+            array(
+                'apiUrl' => 'http://87.98.188.77:9193',
```
Comment (jeus) [inline tests/PaymentGatewayCreateTest.php:43]: remove all host and URL from the code.

**Comment by jeus** [comment]
solving the problems and request again.

File: tests/PaymentGatewayConstructorTest.php (hunk 5 above)
Comment (mohsentm) [reply tests/PaymentGatewayConstructorTest.php:156]: this is just sample url  for test constructor validation url

File: tests/PaymentGatewayConstructorTest.php (hunk 4 above)
Comment (mohsentm) [reply tests/PaymentGatewayConstructorTest.php:28]: this is just sample mobile  for test constructor validation parameter

File: readme.md (hunk 3 above)
Comment (mohsentm) [reply readme.md:164]: Change to apache 2

File: composer.json (hunk 2 above)
Comment (mohsentm) [reply composer.json:20]: changed url

File: readme.md (hunk 1 above)
Comment (mohsentm) [reply readme.md:23]: our register api form exist of this url

**Pull request closed without merging**